### ISSUE TITLE
UIAutomation keeps Form instance alive

### ISSUE THREAD

**Issue by miloszkukla** [dotnet/winforms#660]
* Tested on .NET Framework 4.5.2 only.

**Problem description:**

Obtaining a Form's AutomationElement causes Form to be not eligible for garbage collection - Form instances remain in memory - this is illustrated in the attached repro. This causes trouble for creating stress tests.

This is also reproducible with UIAv3 using wrapper from https://www.nuget.org/packages/TestStack.White.Uia3/ package.

I didn't check if the same behavior exists for WPF Window.

**Actual behavior:**

Form is not eligible for garbage collection until end of method is reached.

**Expected behavior:**

Form is eligible for garbage collection as soon as AutomationElement pointing to the Form goes out of the scope.

**Minimal repro:**

[WindowsFormsApp4.zip](https://github.com/dotnet/winforms/files/3029300/WindowsFormsApp4.zip)

**Comment by miloszkukla** [comment]
After more experimentation it seems that Form actually is garbage collected but only after execution goes out of the method - (counter-intuitively as Form's AutomationElement  goes out of scope after each iteration) - I guess it happens due AutomationElement.Root is caching found descendant AutomationElements. After moving AutomationElement.Root declaration to separate method which is invoked in the loop the results are better (only one alive instance left [don't know why one, not zero]) - but AutomationElement.Root starts to behave weird - FindFirst fails to find Form2 unless I put AutomationElemment.FindAll before it.

Besides GC.WaitForFullGCComplete is not what I thought it is so it usage doesn't make sense.
